objects/src/accounts/builder/mod.rs

```diff
@@ -0,0 +1,313 @@
+        self.account_type = account_type;
+        self
+    }
+
+    /// Sets the storage mode of the account.
+    pub fn storage_mode(mut self, storage_mode: AccountStorageMode) -> Self {
+        self.storage_mode = storage_mode;
+        self
+    }
+
+    /// Sets the nonce of the account. This method is optional.
+    ///
+    /// If unset, the nonce will default to [`ZERO`].
+    pub fn nonce(mut self, nonce: Felt) -> Self {
+        self.nonce = nonce;
+        self
+    }
+
+    /// Adds the asset to the account's [`AssetVault`]. This method is optional.
+    pub fn with_asset(mut self, asset: Asset) -> Self {
+        self.assets.push(asset);
+        self
+    }
+
+    /// Adds all the assets to the account's [`AssetVault`]. This method is optional.
+    pub fn with_assets<I: IntoIterator<Item = Asset>>(mut self, assets: I) -> Self {
+        self.assets.extend(assets);
+        self
+    }
+
+    /// Adds an [`AccountComponent`] to the builder. This method can be called multiple times and
+    /// **must be called at least once** since an account must export at least one procedure.
+    ///
+    /// All components will be merged to form the final code and storage of the built account.
+    pub fn with_component(mut self, account_component: impl Into<AccountComponent>) -> Self {
+        self.components.push(account_component.into());
+        self
+    }
+
+    /// Builds an [`Account`] out of the configured builder.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - The init seed is not set.
+    /// - If a duplicate assets was added to the builder.
+    /// - Any of the components does not support the set account type.
+    /// - The number of procedures in all merged components is 0 or exceeds
+    ///   [`AccountCode::MAX_NUM_PROCEDURES`](crate::accounts::AccountCode::MAX_NUM_PROCEDURES).
+    /// - Two or more libraries export a procedure with the same MAST root.
+    /// - The number of [`StorageSlot`](crate::accounts::StorageSlot)s of all components exceeds
+    ///   255.
+    /// - [`MastForest::merge`](vm_processor::MastForest::merge) fails on the given components.
+    pub fn build(self) -> Result<(Account, Word), AccountBuildError> {
+        let init_seed = self.init_seed.ok_or(AccountBuildError::AccountInitSeedNotSet)?;
+
+        let vault = AssetVault::new(&self.assets).map_err(AccountBuildError::AssetVaultError)?;
+
+        let (code, storage) =
+            Account::initialize_from_components(self.account_type, &self.components)
+                .map_err(AccountBuildError::ComponentInitializationError)?;
+
+        let code_commitment = code.commitment();
+        let storage_commitment = storage.commitment();
+
+        let seed = AccountId::get_account_seed(
+            init_seed,
+            self.account_type,
+            self.storage_mode,
+            code_commitment,
+            storage_commitment,
+        )
+        .map_err(AccountBuildError::AccountSeedGenerationFailure)?;
```

Comment:
This account ID generation process is relevant only for new accounts (i.e., `nonce == ZERO`). Technically, it is not wrong to do this for existing accounts as well - but currently this also wastes a lot of PoW when it doesn't need to.

Reply:
Added a `build_testing` function which should be used in testing code which will not grind an account id for existing accounts.